### ISSUE TITLE
Incorrect Base Requirements for installing Cilium

### ISSUE THREAD

**Issue by RonBarkan** [cilium/cilium#36600]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

equal or higher than v1.16.1 and lower than v1.17.0

### What happened?

The [Base Kernel Config Requirements](https://docs.cilium.io/en/stable/operations/system_requirements/#base-requirements) for installing Cilium seem to be missing config options.

For us, installing Cilium failed with these options missing: 
* `CONFIG_XFRM` was required to not get `creating netlink handle: protocol not supported` error from the cilium-agent, without configuring IPSec. See more here https://github.com/cilium/cilium/discussions/36478
* `CONFIG_VXLAN` for VXLAN, otherwise Cilium fails to start with an error `creating vxlan device: creating device cilium_vxlan: operation not supported, maybe kernel module for vxlan is not available`

### How can we reproduce the issue?

1. Install a Linux Kernel without these config options
2. Install Kubernetes using the [official instructions](https://v1-29.docs.kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/) without a package manager
3. A few days later :-), once the above works, install Cilium using helm charts

### Cilium Version

1.16.1

### Kernel Version

6.1.82-yocto-standard-icebreaker-1f996a3 x86_64 GNU/Linux

### Kubernetes Version

1.29.6

### Regression

I don't know

### Sysdump

N/A

### Relevant log output

```shell
See Above
```

### Cilium Users Document

- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by julianwiedmann** [comment]
The XFRM part makes me think that maybe we introduced some cleanup logic (for the case of IPsec being switched from `enabled` to `disabled`). And that logic not tolerating such errors (if XFRM is not enabled, clearly there's nothing to clean up).

**Comment by RonBarkan** [comment]
`netlink.NewHandle()`  is used for something [here](https://github.com/cilium/cilium/blob/eaefb04be57994037f346193fd63c4e65e23b33a/pkg/datapath/linux/devices_controller.go#L717). It will try to use one of these [families](https://github.com/vishvananda/netlink/blob/main/nl/nl_linux.go#L38).

Perhaps this requirement is due to the switch to  `vishvananda/netlink`?

**Comment by julianwiedmann** [comment]
> `netlink.NewHandle()` is used for something [here](https://github.com/cilium/cilium/blob/eaefb04be57994037f346193fd63c4e65e23b33a/pkg/datapath/linux/devices_controller.go#L717). It will try to use one of these [families](https://github.com/vishvananda/netlink/blob/main/nl/nl_linux.go#L38).
> 
> Perhaps this requirement is due to the switch to `vishvananda/netlink`?

yep good catch, I think that's exactly what's causing it. Let's see what https://github.com/cilium/cilium/pull/37123 brings, maybe you want to give that a try?

**Comment by julianwiedmann** [comment]
>CONFIG_VXLAN for VXLAN, otherwise Cilium fails to start with an error creating vxlan device: creating device cilium_vxlan: operation not supported, maybe kernel module for vxlan is not available

Could you maybe open a PR for this to improve the docs? For `routingMode=tunnel` (which is the default) we indeed need either `CONFIG_VXLAN` or `CONFIG_GENEVE` (depending on the selected protocol, VXLAN again being the default).

**Comment by julianwiedmann** [comment]
> > CONFIG_VXLAN for VXLAN, otherwise Cilium fails to start with an error creating vxlan device: creating device cilium_vxlan: operation not supported, maybe kernel module for vxlan is not available
> 
> Could you maybe open a PR for this to improve the docs? For `routingMode=tunnel` (which is the default) we indeed need either `CONFIG_VXLAN` or `CONFIG_GENEVE` (depending on the selected protocol, VXLAN again being the default).

I believe this is now addressed with https://github.com/cilium/cilium/pull/38173. Closing here.
